src/vivarium_census_prl_synth_pop/components/businesses.py

```diff
@@ -106,9 +124,19 @@ def on_initialize_simulants(self, pop_data: SimulantData) -> None:
             working_age = pop.loc[pop["age"] >= data_values.WORKING_AGE].index
             pop.loc[working_age, "employer_id"] = self.assign_random_employer(working_age)
 
+            # Create income propensity columns
+            pop["personal_income_propensity"] = self.randomness.get_draw(
+                pop.index,
+                additional_key="personal_income_propensity",
+            )
+            pop["employer_income_propensity"] = self.randomness.get_draw(
+                pop.index,
+                additional_key="employer_income_propensity",
+            )
+
             # merge on employer addresses and names
             pop = pop.merge(
-                self.businesses[self.columns_created[:-1]], on="employer_id", how="left"
+                self.businesses[self.columns_created[:3]], on="employer_id", how="left"
```

Comment:
This seems like a brittle change.

Reply:
I agree. It's much better to name the columns than refer to them by index like this. If you want to exclude specific columns from `self.columns_created` you should do that explicitly.